### PR TITLE
Override password reset email send when creating user if SMTP not set up

### PR DESCRIPTION
**Pull request by jhs-panda**
## Description

If SMTP is not properly set up, creating a user will not send the typical password reset email. Instead, it will create the user but flash a message to the user.

## Motivation and Context

With SMTP Config set up with single-sender, the password reset email does not send successfully when creating a new user. Issue #2204  
In the case that SMTP is not set up successfully, a user is still created but a message is flashed that the password reset email was not successfully sent. 

## How Has This Been Tested?

Before:
Creating new user
Screen Shot 2024-09-15 at 2 06 51 PM

After:
Creating a user but SMTP not set up.
Screen Shot 2024-09-29 at 2 08 28 PM
Screen Shot 2024-09-29 at 2 08 45 PM

If error other than SMTP comes up, flash error message and destroy user (since already created but if unusable, no point in keeping).
Screen Shot 2024-09-29 at 2 10 04 PM

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run rubocop and erblint for style check. If you haven't, run `overcommit --install && overcommit --sign` to use pre-commit hook for linting
- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://docs.autolabproject.com/)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

**Comment by 20wildmanj** [comment]
Could you change this so that instead of just commenting out the send password reset lines, there is code to catch the exception and flash the error?

**Comment by KesterTan** [comment]
I'm getting this error with this config

Screenshot 2024-10-06 at 12 40 39 PM

Screenshot 2024-10-06 at 12 43 17 PM

Screenshot 2024-10-06 at 12 43 24 PM

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -117,9 +117,19 @@ def create
...
-      redirect_to(users_path) && return
+      begin
+        @user.send_reset_password_instructions
+        flash[:success] = "Successfully created user."
+      rescue Net::SMTPFatalError
+        flash[:success] = "Successfully created user but reset password instructions were not sent:
+          Net::SMTPFatalError"
+      rescue StandardError => e
+        error_message = e.message
+        flash[:error] = "Failed to create user: #{error_message}"
```
Comment (KesterTan) [inline app/controllers/users_controller.rb:128]: This flash error is slightly confusing to me, I've been getting error messages like localhost(25) and auth errors when SMTP is not configured and I don't think I would know it was related to my SMTP config unless I know that an email was trying to be sent. Could we change this error message to say something like `Failed to create user: incorrectly configured SMTP config : #{error_message}` or something.

Comment (jhs-panda) [reply app/controllers/users_controller.rb]: Done.

**Review by KesterTan — APPROVED** [review]
LGTM

**Pull request merged**